Dockerfile

```diff
@@ -1,37 +1,47 @@
 ### Build image ###
-FROM nvidia/cudagl:11.4.2-devel-ubuntu20.04 as build
+FROM nvidia/cuda:12.1.1-devel-ubuntu22.04 as build
 
 # install dependencies
 ARG DEBIAN_FRONTEND=noninteractive
-RUN mkdir /agave && \
-    mkdir /agave/build && \
-    apt-get update && apt-get install -y \
+RUN apt-get update && apt-get install -y \
     apt-utils \
     build-essential \
+    software-properties-common \
     git \
     wget \
-    libspdlog-dev \
-    libtiff-dev \
-    libglm-dev \
-    libgl1-mesa-dev \
-    libegl1-mesa-dev \
     libgles2-mesa-dev \
     libegl1 \
     xvfb \
     xauth \
+    libspdlog-dev \
+    libglm-dev \
+    libgl1-mesa-dev \
+    libegl1-mesa-dev \
+    libtiff-dev \
     libzstd-dev \
-    nasm
+    nasm \
+    libglvnd0 \
+    libgl1 \
+    libglx0 \
+    libegl1 \
+    libgles2 \
+    libxkbcommon-x11-0
 
+RUN apt-get install -y apt-transport-https ca-certificates gnupg
+# get gcc-10 (not default on ubuntu 20.04)
+RUN add-apt-repository -y ppa:ubuntu-toolchain-r/test
+RUN apt-get install -y gcc-10 g++-10
+#RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-11 11
+#RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 11
 # get a current cmake
-RUN apt-get update && apt-get install -y apt-transport-https ca-certificates gnupg software-properties-common
 RUN wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /etc/apt/trusted.gpg.d/kitware.gpg >/dev/null
-RUN apt-add-repository 'deb https://apt.kitware.com/ubuntu/ focal main'
+RUN apt-add-repository 'deb https://apt.kitware.com/ubuntu/ jammy main'
 RUN apt-get install kitware-archive-keyring
 RUN rm /etc/apt/trusted.gpg.d/kitware.gpg
 RUN apt-get update && apt-get install -y cmake
 
 # get python
-RUN apt-get install -y python3.9-dev python3-pip
+RUN apt-get install -y python3-pip
```

Comment:
So the default python is fine now?

Reply:
the base image is now ubuntu 22 which has python 3.10 as default